src/java.base/share/classes/java/net/URLEncoder.java

```diff
@@ -204,6 +204,9 @@ public static String encode(String s, String enc)
      * "http://www.w3.org/TR/html40/appendix/notes.html#non-ascii-chars">
      * World Wide Web Consortium Recommendation</a> states that
      * UTF-8 should be used. Not doing so may introduce incompatibilities.</em>
+     * <p>
+     * If a character needs encoding but cannot be encoded, the
+     * {@linkplain CharsetEncoder##cae replacement bytes} will be used.
```

Comment:
I think this text will appear in the "Note" section of the method description. We are adding normative text so I think would be better if the new text went into the first paragraph or introduce a new parameter before the "Note". We could replace the "Note" heading with `@apiNote` if you want to clean this up.

As regards the text,  I think it would be more correct to say that if the input string is malformed, or if the input cannot be mapped to a valid byte sequence in the given charset, then the erroneous input with be replaced with the charset's replacement value.